indexprovider/wrapper.go

```diff
@@ -0,0 +1,136 @@
+package indexprovider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/boost/storagemarket/types/dealcheckpoints"
+	"github.com/hashicorp/go-multierror"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/boost/storagemarket/types"
+	metadata2 "github.com/filecoin-project/index-provider/metadata"
+	"github.com/filecoin-project/lotus/markets/dagstore"
+	"github.com/filecoin-project/lotus/markets/idxprov"
+
+	"github.com/filecoin-project/boost/db"
+	lotus_storagemarket "github.com/filecoin-project/go-fil-markets/storagemarket"
+	provider "github.com/filecoin-project/index-provider"
+	"github.com/ipfs/go-cid"
+)
+
+var log = logging.Logger("index-provider-wrapper")
+
+type Wrapper struct {
+	dealsDB     *db.DealsDB
+	legacyProv  lotus_storagemarket.StorageProvider
+	prov        provider.Interface
+	dagStore    *dagstore.Wrapper
+	meshCreator idxprov.MeshCreator
+}
+
+func NewWrapper(dealsDB *db.DealsDB, legacyProv lotus_storagemarket.StorageProvider, prov provider.Interface, dagStore *dagstore.Wrapper,
+	meshCreator idxprov.MeshCreator) *Wrapper {
+	return &Wrapper{
+		dealsDB:     dealsDB,
+		legacyProv:  legacyProv,
+		prov:        prov,
+		dagStore:    dagStore,
+		meshCreator: meshCreator,
+	}
+}
+
+func (w *Wrapper) IndexerAnnounceAllDeals(ctx context.Context) error {
```

Comment:
We can do it or we can do it as we're currently doing it. I don't have a strong preference.

Reply:
Where possible I'd prefer to keep knowledge of the two different data sources inside the wrapper, rather than leaking it into the CLI layer

Reply:
Man, moving all that markets code here (which has intimate knowledge of various market states and what they mean) will add mantainence burden here and is something I'd like to avoid. Remember, the plan is to eventually deprecate markets.